devito/types/dimension.py

```diff
@@ -1095,6 +1095,11 @@ def symbolic_incr(self):
         except (TypeError, ValueError):
             return self.step
 
+    @cached_property
+    def level(self):
+        # The level of an `IncrDimension` in the hierarchy of `IncrDimension`s
+        return len([i for i in self._defines if i.is_Incr])
```

Comment:
the level should trivially be defined as the depth in the hierarchy, and it could well be a DerivedDimension property (and Dimension.level returning 0 as Mathias was suggesting above)